src/Sylius/Bundle/ShopBundle/Controller/ContactController.php

```diff
@@ -0,0 +1,156 @@
+     * @param ChannelContextInterface $channelContext
+     * @param CustomerContextInterface $customerContext
+     * @param ContactEmailManager $contactEmailManager
+     */
+    public function __construct(
+        RouterInterface $router,
+        FormFactoryInterface $formFactory,
+        EngineInterface $templatingEngine,
+        ChannelContextInterface $channelContext,
+        CustomerContextInterface $customerContext,
+        ContactEmailManager $contactEmailManager
+    ) {
+        $this->router = $router;
+        $this->formFactory = $formFactory;
+        $this->templatingEngine = $templatingEngine;
+        $this->channelContext = $channelContext;
+        $this->customerContext = $customerContext;
+        $this->contactEmailManager = $contactEmailManager;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function requestAction(Request $request)
+    {
+        $formType = $this->getSyliusAttribute($request, 'form', ContactType::class);
+        $form = $this->formFactory->create($formType, null, $this->getFormOptions());
+
+        if ($request->isMethod('POST') && $form->handleRequest($request)->isValid()) {
+            $data = $form->getData();
+            $channel = $this->channelContext->getChannel();
+            $contactEmail = $channel->getContactEmail();
+
+            if (null === $contactEmail) {
+                $errorMessage = $this->getSyliusAttribute($request, 'error_flash', 'sylius.contact.request_error');
+                $request->getSession()->getFlashBag()->add('error', $errorMessage);
+
+                return new RedirectResponse($request->headers->get('referer'));
+            }
+
+            $this->contactEmailManager->sendContactRequest($data, [$contactEmail]);
+
+            $successMessage = $this->getSyliusAttribute($request, 'success_flash', 'sylius.contact.request_success');
+            $request->getSession()->getFlashBag()->add('success', $successMessage);
+
+            $redirectRoute = $this->getSyliusAttribute($request, 'redirect', 'referer');
+
+            return new RedirectResponse($this->router->generate($redirectRoute));
+        }
+
+        $template = $this->getSyliusAttribute($request, 'template', "@SyliusShop/Contact/request.html.twig");
+
+        return $this->templatingEngine->renderResponse(
+            $template,
+            [
+                'form' => $form->createView(),
+            ]
+        );
+    }
+
+    /**
+     * @param Request $request
+     * @param string $attributeName
+     * @param string|null $default
+     *
+     * @return string|null
+     */
+    private function getSyliusAttribute(Request $request, $attributeName, $default = null)
+    {
+        $attributes = $request->attributes->get('_sylius');
+
+        return isset($attributes[$attributeName]) ? $attributes[$attributeName] : $default;
+    }
+
+    /**
+     * @return array
+     */
+    private function getFormOptions()
```

Comment:
IMO `getFormOptions` is a better name for this method, because it prepares options for a form..